Installing cargo-lambda version 0.14.0 fails.

While installing the cargo-lambda package with the command: `cargo install cargo-lambda` (which resolves to 0.14.0 at the current point in time)

It fails when compiling the cargo-lambda-watch crate.
```rust
Compiling cargo-lambda-watch v0.14.0
error[E0308]: mismatched types
   --> /Users/dkMaHoJe/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-lambda-watch-0.14.0/src/watcher.rs:65:40
    |
65  |             .add_globs(&["**/*"], Some(wc.base))
    |                                   ---- ^^^^^^^
    |                                   |    |
    |                                   |    expected `&PathBuf`, found struct `PathBuf`
    |                                   |    help: consider borrowing here: `&wc.base`
    |                                   arguments to this enum variant are incorrect
    |
note: tuple variant defined here
   --> /Users/dkMaHoJe/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/option.rs:526:5
    |
526 |     Some(#[stable(feature = "rust1", since = "1.0.0")] T),
    |     ^^^^

error[E0277]: `Result<(), ignore_files::Error>` is not a future
  --> /Users/dkMaHoJe/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-lambda-watch-0.14.0/src/watcher.rs:66:13
   |
66 |             .await
   |             ^^^^^^
   |             |
   |             `Result<(), ignore_files::Error>` is not a future
   |             help: remove the `.await`
   |
   = help: the trait `Future` is not implemented for `Result<(), ignore_files::Error>`
   = note: Result<(), ignore_files::Error> must be a future or must implement `IntoFuture` to be awaited
   = note: required for `Result<(), ignore_files::Error>` to implement `IntoFuture`

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `cargo-lambda-watch` due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error: failed to compile `cargo-lambda v0.14.0`, intermediate artifacts can be found at `/var/folders/lr/sq4c_jm56k51tdjj57gj2dc00000gp/T/cargo-installULa483`
```

This is a snippet of the full install log:
https://gist.github.com/MagnusHJensen/8ad89c5425382198d6438b38d01c20bd

Here is our cargo.toml file if that helps.
https://gist.github.com/MagnusHJensen/7c94f4b0e40ca1a8c85d1207a7f592ba

Before running this command, we also run the `pip3 install ziglang`


Feel free to let me know if you need further information.
This issue occurs on my machine (Macbook Pro 2021) but it also happens inside github actions running latest ubuntu (22.04) I think.

--- a/crates/cargo-lambda-watch/Cargo.toml
+++ b/crates/cargo-lambda-watch/Cargo.toml
@@ -27,7 +27,7 @@ clap.workspace = true
 dunce.workspace = true
 http-api-problem = { version = "0.51.0", features = ["api-error", "hyper"] }
 hyper = "0.14.18"
-ignore-files = "1.0.1"
+ignore-files = "=1.1.0"
 miette.workspace = true
 opentelemetry = "0.17.0"
 opentelemetry-aws = "0.5.0"
@@ -47,5 +47,5 @@ tower-http = { version = "0.3.3", features = [
 tracing.workspace = true
 tracing-opentelemetry = "0.17.2"
 tracing-subscriber.workspace = true
-watchexec = "2.0.2"
+watchexec = "=2.1.0"
 watchexec-filterer-ignore = "1.0.0"

--- a/crates/cargo-lambda-watch/src/watcher.rs
+++ b/crates/cargo-lambda-watch/src/watcher.rs
@@ -62,8 +62,7 @@ async fn runtime(cmd: Command, wc: WatcherConfig) -> Result<RuntimeConfig, Serve
         .map_err(ServerError::InvalidIgnoreFiles)?;
     if wc.no_reload {
         filter
-            .add_globs(&["**/*"], Some(wc.base))
-            .await
+            .add_globs(&["**/*"], Some(&wc.base))
             .map_err(ServerError::InvalidIgnoreFiles)?;
     }
     config.filterer(Arc::new(watchexec_filterer_ignore::IgnoreFilterer(filter)));

